### PR TITLE
NOISSUE - Raise endpoint test coverage

### PR DESCRIPTION
**Pull request by dborovcanin**
Cover some edge cases and fix decodeList to always return same error.

Signed-off-by: Dušan Borovčanin <borovcanindusan1@gmail.com>

### CODE REVIEW

**Review by drasko — COMMENTED** [review]
LGTM

File: manager/api/transport.go [hunk 1]
```diff
@@ -278,21 +278,21 @@ func decodeList(_ context.Context, r *http.Request) (interface{}, error) {
 
 	off, lmt := q["offset"], q["limit"]
 
-	if len(off) > 1 || len(lmt) > 1 {
+	if len(off) != 1 || len(lmt) != 1 {
 		return nil, errInvalidQueryParams
 	}
 
 	if len(off) == 1 {
```
Comment (mijicd) [inline manager/api/transport.go]: Most likely it isn't 1, based on the check added above :)

File: manager/api/transport.go [hunk 2]
```diff
@@ -278,21 +278,21 @@ func decodeList(_ context.Context, r *http.Request) (interface{}, error) {
...
 		offset, err = strconv.Atoi(off[0])
 		if err != nil {
-			return nil, err
+			return nil, errInvalidQueryParams
 		}
 	}
 
 	if len(lmt) == 1 {
```
Comment (mijicd) [inline manager/api/transport.go:277]: Same here.

File: manager/api/endpoint_test.go [hunk 3]
```diff
@@ -347,21 +347,53 @@ func TestListClients(t *testing.T) {
...
+		desc   string
+		status int
+		URL    string
+		res    []manager.Client
+	}{
+		{"get a list of clients with default URL", http.StatusOK, fmt.Sprintf("%s%s", clientURL, ""), clients[0:10]},
+		{"get a list of clients with invalid URL", http.StatusBadRequest, fmt.Sprintf("%s%s", clientURL, "?%%"), nil},
+		{"get a list of clients with invalid query params", http.StatusBadRequest, fmt.Sprintf("%s%s", clientURL, "?limit=no_limit&a=a&b=b"), nil},
+		{"get a list of clients with invalid number of params", http.StatusBadRequest, fmt.Sprintf("%s%s", clientURL, "?offset=4&limit=4&limit=5&offset=5"), nil},
+		{"get a list of clients with invalid offset", http.StatusBadRequest, fmt.Sprintf("%s%s", clientURL, "?offset=e&limit=5"), nil},
```
Comment (mijicd) [inline manager/api/endpoint_test.go]: You might use the edge cases (negative or zero) as test case as well.

File: manager/api/endpoint_test.go [hunk 4]
```diff
@@ -569,19 +601,52 @@ func TestListChannels(t *testing.T) {
...
+
+	cases := []struct {
+		desc   string
+		status int
+		URL    string
+		res    []manager.Channel
+	}{
+		{"get a list of channels with default URL", http.StatusOK, fmt.Sprintf("%s%s", channelURL, ""), channels[0:10]},
+		{"get a list of channels with invalid URL", http.StatusBadRequest, fmt.Sprintf("%s%s", channelURL, "?%%"), nil},
+		{"get a list of channels with invalid query params", http.StatusBadRequest, fmt.Sprintf("%s%s", channelURL, "?limit=no_limit&a=a&b=b"), nil},
```
Comment (mijicd) [inline manager/api/endpoint_test.go]: Same here.

**Review by mijicd — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
@dusanb94 please also rebase to latest master

File: manager/api/transport.go [hunk 5]
```diff
@@ -281,14 +266,14 @@ func decodeList(_ context.Context, r *http.Request) (interface{}, error) {
 	if len(off) == 1 {
```
Comment (anovakovic01) [inline manager/api/transport.go:270]: Maybe you should extract these checks and conversion to int into separate function. It is repeated two times here.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]

**Review by mijicd — APPROVED** [review]
